akka-actor-tests/src/test/scala/akka/event/EventBusSpec.scala

```diff
@@ -11,39 +11,51 @@ import akka.testkit._
 import scala.concurrent.duration._
 
 import akka.actor.{ Actor, ActorRef, ActorSystem, PoisonPill, Props }
+import akka.actor.actorRef2Scala
 import akka.japi.{ Procedure }
 import com.typesafe.config.{ Config, ConfigFactory }
 
 object EventBusSpec {
+  type EventBusAux[E, C, S] = EventBus {
+    type Event = E
+    type Classifier = C
+    type Subscriber = S
+  }
   class TestActorWrapperActor(testActor: ActorRef) extends Actor {
     def receive = {
       case x => testActor.forward(x)
     }
   }
 }
 
-abstract class EventBusSpec(busName: String, conf: Config = ConfigFactory.empty())
+import EventBusSpec._
+abstract class EventBusSpec[
+    Event,
+    Classifier,
+    Subscriber,
+    BusType <: EventBusSpec.EventBusAux[Event, Classifier, Subscriber]](
+    busName: String,
+    conf: Config = ConfigFactory.empty())
```

Comment:
yep, the refactoring for type projections will be intrusive. It would be nice to explore some alternatives to reduce the impact on users. However, that won't be zero-impact.